src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HPack/DynamicTable.cs

```diff
@@ -34,7 +34,14 @@ public DynamicTable(int maxSize)
                     throw new IndexOutOfRangeException();
                 }
 
-                return _buffer[_insertIndex == 0 ? _buffer.Length - 1 : _insertIndex - index - 1];
+                index = _insertIndex - index - 1;
+
+                if (index < 0)
+                {
+                    index += _buffer.Length;
+                }
```

Comment:
Can you add a comment about what this is doing?  Looks like it's treating the buffer as circular?

Reply:
Correct. Will do.